State_manager_code/StateManager.py

```diff
@@ -37,12 +37,11 @@ def __init__(self):
             States.Start : [States.CenterReward],
             States.CenterReward : [States.TrialStarted],
             States.TrialStarted : [States.M1CM2C, States.M1CM2D, States.M1DM2C, States.M1DM2D],
-            States.M1CM2C : [States.WaitForReturn],
-            States.M1CM2D: [States.WaitForReturn],
-            States.M1DM2C: [States.WaitForReturn],
-            States.M1DM2D: [States.WaitForReturn],
-            States.WaitForReturn:[States.TrialCompleted],
-            States.TrialCompleted:[States.End],
+            States.M1CM2C : [States.TrialCompleted],
+            States.M1CM2D: [States.TrialCompleted],
+            States.M1DM2C: [States.TrialCompleted],
+            States.M1DM2D: [States.TrialCompleted],
+            States.TrialCompleted:[States.CenterReward,States.End],
             States.TrialAbort:[States.End, States.TrialStarted],
```

Comment:
There is no need for TrialEnd transition in the TrialAbort (a.k.a known as ReturnTimeOut state). We are entering this state only from TrialCompleted. So is the last Trial has been reached it will exit before this state.

Reply:
As I said you are right and we do need it. Take a look at the latest state diagram